apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/PublisherClient.cs

```diff
@@ -634,11 +658,22 @@ public override void ResumePublish(string orderingKey)
             }
         }
 
+        /// <inheritdoc/>        
+        public override ValueTask DisposeAsync() => new ValueTask(ShutdownAsync(_disposeTimeout));
+
         /// <inheritdoc/>
         public override Task ShutdownAsync(CancellationToken hardStopToken)
         {
             lock (_lock)
             {
+                // If we come here for a second or subsequent time, this condition would always be true.
+                // Note: If multiple shutdown requests are made, only the first cancellation token is observed. 
+                if (_softStopCts.IsCancellationRequested)
+                {
+                    // No-op. We don't want to throw exceptions if DisposeAsync or ShutdownAsync is called a second time.
+                    return _shutdownTcs.Task;
+                }
+
```

Comment:
Please remove the CheckShutdown call in the subsequent line - it's designed to throw if we call this twice, which the code above already handles by returning the existing task, so it's pointless.

Reply:
Right. Done.